propsigner/propsigner_test.go

```diff
@@ -0,0 +1,157 @@
+package propsigner
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/ipfs/go-cid"
+	libwal "github.com/jsign/go-filsigner/wallet"
+	"github.com/libp2p/go-libp2p-core/peer"
+	swarmt "github.com/libp2p/go-libp2p-swarm/testing"
+	bhost "github.com/libp2p/go-libp2p/p2p/host/basic"
+	"github.com/stretchr/testify/require"
+	"github.com/textileio/go-auctions-client/localwallet"
+)
+
+var (
+	walletKeys = []string{
+		// Secp256k1 exported private key in Lotus format.
+		"7b2254797065223a22736563703235366b31222c22507269766174654b6579223a226b35507976337148327349586343595a58594f5775453149326e32554539436861556b6c4e36695a5763453d227d", // nolint:lll
+		// BLS exported private key in Lotus format.
+		"7b2254797065223a22626c73222c22507269766174654b6579223a226862702f794666527439514c43716b6d566171415752436f50556777314b776971716e73684e49704e57513d227d", // nolint:lll
+	}
+)
+
+type testCase struct {
+	name      string
+	proposal  market.DealProposal
+	authToken string
+	err       error
+}
+
+func TestProposalSigning(t *testing.T) {
+	t.Parallel()
+
+	authToken := "veryhardtokentoguess"
+	wallet, err := localwallet.New(walletKeys)
+	require.NoError(t, err)
+
+	testCases := []testCase{
```

Comment:
We test success cases with different types of wallet.
We test failure cases reg wrong auth tokens and wallet addresses that we don't have the keys.